### ISSUE TITLE
Question on packaging

### ISSUE THREAD

**Issue by gabrielrodcanal** [nqdtan/vck5000_vivado_custom_ulp_design#4]
Hi,
I have observed that in order to be able to run a different kernel on the AIE it is necessary to both rebuild both the ELF `core.out` and the `.xclbin`. Could you please explain how these two files are connected? I don't see any reference to `core.out` in the `xclbin_generator` with AIEs and `core.out` seems to be loaded dynamically here: https://github.com/nqdtan/vck5000_vivado_ulp/blob/d78ccee4d0e1ca3556e4b96fe12df145dde5ca1e/host_sw_with_aie/host.cpp#L95

**Comment by nqdtan** [comment]
Yes. `core.out` is loaded dynamically in the host code via `XAie_LoadElf` at runtime. `XAie_LoadElf` converts the core binary to a stream of 32-bit configuration data, and the `data_mover` IP sends it to the AIE tile over the NoC. There should be no connection with `xclbin`. How did you rebuild `xclbin`? Did you change something in it or just simply reload the bitstream? I think the reason is that the host code might not properly reset some AIE config between different runs.

I found that after using the card for a while, I have to reboot the machine to make sure the AIE is fully reset to the initial state (all the configs are cleared) when testing new code. Another approach that doesn't require rebooting the host machine is to have another "dummy" bitstream that does not use AIE; when you want to reset the AIE state, you can load that bitstream, and then reload your `ulp.xclbin`. There are some XAie functions that help clear the AIE config as well. The host code here is by no means comprehensive.
